features/parameter-types/rooms.js

```diff
@@ -1,4 +1,6 @@
 const { defineParameterType } = require("cucumber");
+const concatRegExp = require('../utilities/concatRegExp');
+const FLEXIBLE_ARTICLE_ADJECTIVES = /(an |the |is |a )/;
```

Comment:
Not blocking: The union of these two pieces feels like a meaningful/useful part of the program; as they are used together.

Perhaps it's `withArticleAdjectives` and it's used like:

`regexp: withArticleAdjectives(/(Unlocked|Internal|Locked)/)` or `regexp: withArticleAdjectives(/( |correct |wrong )?Room Key/)` etc?